lib/Sema/CSGen.cpp

```diff
@@ -2062,48 +2062,38 @@ namespace {
       // parameter or return type is omitted, a fresh type variable is used to
       // stand in for that parameter or return type, allowing it to be inferred
       // from context.
-      auto getExplicitResultType = [&]() -> Type {
-        if (!closure->hasExplicitResultType()) {
-          return Type();
-        }
+      Type resultTy = [&] {
+        if (closure->hasExplicitResultType()) {
+          if (auto declaredTy = closure->getExplicitResultType()) {
+            return declaredTy;
+          }
 
-        if (auto declaredTy = closure->getExplicitResultType()) {
-          return declaredTy;
+          const auto resolvedTy = resolveTypeReferenceInExpression(
+              closure->getExplicitResultTypeRepr(),
+              TypeResolverContext::InExpression,
+              // Introduce type variables for unbound generics.
+              OpenUnboundGenericType(
```

Comment:
Could you please split this into two commits - first one to add `OpenUnboundGenericType` to the original code which is an actual functionality change here and second which does this `resultTy` refactoring?

Reply:
Easy